File: src/test/java/model/util/PositionIteratorTest.java
package model.util;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class PositionIteratorTest {
    @Test
    void test() {
        PositionIterator positionIterator = new PositionIterator();

        int i = 0;

        while (positionIterator.hasNext()) {
            Position position = positionIterator.next();

            if (i == 0) {
                Assertions.assertEquals(new Position(0, 0), position);
            } else if (i == 1) {
                Assertions.assertEquals(new Position(0, 1), position);
            } else if (i == Position.LIMITE * Position.LIMITE - 1) {
                Assertions.assertEquals(new Position(Position.LIMITE - 1, Position.LIMITE - 1), position);
            } else if (i >= Position.LIMITE * Position.LIMITE) {
                throw new RuntimeException("Position trop grande: " + position.toString());
            }

            i++;
        }
    }
}

File: engine/src/test/java/engine/util/PositionIteratorTest.kt
package engine.util

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class PositionIteratorTest {
    @Test
    fun test() {
        val positionIterator = PositionIterator()

        var i = 0

        while (positionIterator.hasNext()) {
            val position = positionIterator.next()

            when {
                i == 0 -> Assertions.assertEquals(Position(0, 0), position)
                i == 1 -> Assertions.assertEquals(Position(0, 1), position)
                i == Position.LIMIT * Position.LIMIT - 1 -> Assertions.assertEquals(Position(Position.LIMIT - 1, Position.LIMIT - 1), position)
                i >= Position.LIMIT * Position.LIMIT -> throw RuntimeException("Position is too large: " + position.toString())
            }

            i++
        }
    }
}